packages/legacy/core/App/utils/parsers.tsx

```diff
@@ -0,0 +1,111 @@
+export type ParseInvitationResult =
+  | {
+      success: true
+      result: ParsedInvitation
+    }
+  | {
+      success: false
+      error: string
+    }
+
+export type ParsedInvitation = {
+  type: 'didcomm' | 'openid-credential-offer' | 'openid-authorization-request'
+  format: 'url' | 'parsed'
+  data: string | Record<string, unknown>
+}
+
+export enum InvitationQrTypes {
+  OPENID_INITIATE_ISSUANCE = 'openid-initiate-issuance://',
+  OPENID_CREDENTIAL_OFFER = 'openid-credential-offer://',
+  // TODO: I think we should not support openid://, as we mainly support openid4vp
+  // But older requests do use openid:// I think (such as the DIIP dbc login)
+  // but I think we're going to move to just openid4p in the future
+  OPENID = 'openid://',
+  OPENID4VP = 'openid4vp://',
+  OPENID_VC = 'openid-vc://',
+  DIDCOMM = 'didcomm://',
+  HTTPS = 'https://',
+}
+
+export const isOpenIdCredentialOffer = (url: string) => {
+  if (
+    url.startsWith(InvitationQrTypes.OPENID_INITIATE_ISSUANCE) ||
+    url.startsWith(InvitationQrTypes.OPENID_CREDENTIAL_OFFER)
+  ) {
+    return true
+  }
+
+  if (url.includes('credential_offer_uri=') || url.includes('credential_offer=')) {
+    return true
+  }
+
+  return false
+}
+
+export const isOpenIdPresentationRequest = (url: string) => {
+  if (
+    url.startsWith(InvitationQrTypes.OPENID) ||
+    url.startsWith(InvitationQrTypes.OPENID_VC) ||
+    url.startsWith(InvitationQrTypes.OPENID4VP)
+  ) {
+    return true
+  }
+
+  if (url.includes('request_uri=') || url.includes('request=')) {
+    return true
+  }
+
+  return false
+}
+
+export const isDidCommInvitation = (url: string) => {
+  if (url.startsWith(InvitationQrTypes.DIDCOMM)) {
+    return true
+  }
+
+  if (url.includes('c_i=') || url.includes('oob=') || url.includes('oobUrl=') || url.includes('d_m=')) {
```

Comment:
should this not account for `c_i`, `d_m`, `oob`, `_oob`, and `_url`? As it currently checks for those during scan handling

Reply:
@MosCD3 , there was some work done to normalize and remove duplication on how URLs/Payloads are being parsed. We had some duplication and inconsistency between scanning QR Code vs open with deep link.
This is the main entry point:
https://github.com/openwallet-foundation/bifold-wallet/blob/ea2300f5186539d23cc401b90b92cb859f24a9e6/packages/legacy/core/App/utils/helpers.ts#L1000

Reply:
Just to clarify, yes this is the entry point and I need a way to differ open-id invitation from that of did-comm
And the way I did it is through the entry point you mentioned @cvarjao to minimize code impact. I added a parser and the way the parser works is scanning the invitation URL for known did-comm tags in line 66, so the answer @bryce-mcmath is yes we need this to return the type of invitation so that we route the invitation to the correct flow. If you have any other better and more clean flow to have the same result, please suggest it here and I will refactor

Reply:
I think we should just worry about "isOpenIdPresentationRequest()" and then add the logic to handle that, and fallback to what we already have, I would hope that in the future Credo can actually parse the invitation payload and make that decision.

Reply:
this is a nice plan

Reply:
resolved